controllers/wizard/fileUpload/form/PKPSubmissionFilesUploadBaseForm.php

```diff
@@ -217,6 +217,7 @@ public function getSubmissionFiles()
                         ->getCollector()
                         ->filterByReviewRoundIds([(int) $reviewRound->getId()])
                         ->filterBySubmissionIds([$submissionId])
+                        ->filterByFileStages([(int) $this->getData('fileStage')])
                         ->getMany()
```

Comment:
This make sure that we list only from the same "file grid" for the "If you are uploading a revision of an existing file, please indicate which file."